imap_processing/mag/l1a/mag_l1a_data.py

```diff
@@ -429,6 +477,467 @@ def to_signed16(n: int) -> int:
                 secondary_vectors.append(vector)
 
         return (
-            np.array(primary_vectors, dtype=np.int64),
-            np.array(secondary_vectors, dtype=np.int64),
+            np.array(primary_vectors, dtype=np.int32),
+            np.array(secondary_vectors, dtype=np.int32),
         )
+
+    @staticmethod
+    def process_compressed_vectors(  # noqa: PLR0912, PLR0915
+        vector_data: np.ndarray, primary_count: int, secondary_count: int
+    ) -> tuple[np.ndarray, np.ndarray]:
+        """
+        Given raw compressed packet data, process into Vectors.
+
+        To do this, we need to decode the compressed data. The compressed data starts
+        with an 8 bit header that defines the width of the uncompressed vectors and
+        if there is a range data section. Then, the vector data follows, then the range
+        data section if it exists.
+
+        To decode, we start by decoding the first compression_width bits. This is an
+        uncompressed primary vector with range. Then, we proceed through the compressed
+        data, where each value is fibonacci and zig-zag encoded. This means each value
+        ends in 2 sequential ones (11). We split the data along these numbers until
+        we reach primary_count.
```

Comment:
Do you mean this?
```suggestion
        we reach last primary_count.
```

Reply:
This is an awkward sentence, and I will rephrase it, thanks! It should be "until we reach primary_count vectors"